### PR TITLE
Add introduction to Stable RESTT API

### PR DESCRIPTION
**Pull request by mik-laj**
Preview: https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/PolideaInternal/airflow/api-introduction/airflow/api_connexion/openapi/v1.yaml

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/api_connexion/openapi/v1.yaml [hunk 1]
```diff
@@ -19,7 +19,148 @@ openapi: 3.0.3
 
 info:
   title: "Airflow API (Stable)"
-  description: Apache Airflow management API.
+  description: |
+    # Overview
+
+    To enable facilitate management, the Apache Airflow supports a range of REST API endpoints across its
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
    To facilitate the management, the Apache Airflow supports a range of REST API endpoints across its
```

File: airflow/api_connexion/openapi/v1.yaml [hunk 2]
```diff
@@ -19,7 +19,148 @@ openapi: 3.0.3
...
+    objects.
+    This section provides an overview of the API design, methods, and supported use cases.
+
+    Most of the endpoints accept `JSON` as input and return `JSON` responses.
+    This means that you must usually add the following hearders to your request:
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
    This section provides an overview of the API design, methods, and supported use cases.

    Most of the endpoints accept `JSON` as input and return `JSON` responses.
    This means that you must usually add the following headers to your request:
```

File: airflow/api_connexion/openapi/v1.yaml [hunk 3]
```diff
@@ -19,7 +19,148 @@ openapi: 3.0.3
...
+    Accept: application/json
+    ```
+
+    ## Resources
+
+    The term `resource` refers to a single type of object in the Airflow metadata. An API is broken up by its
+    endpoint's corresponding resource.
+    The name of a resource is typically plural, and expressed in camelCase. Example: `jobGroups`.
+
+    Resource names are used as part of endpoint URLs, as well as in API parameters and responses.
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
    The term `resource` refers to a single type of object in the Airflow metadata. An API is broken up by its
    endpoint's corresponding resource.
    The name of a resource is typically plural and expressed in camelCase. Example: `jobGroups`.

    Resource names are used as part of endpoint URLs, as well as in API parameters and responses.
```

File: airflow/api_connexion/openapi/v1.yaml [hunk 4]
```diff
@@ -19,7 +19,148 @@ openapi: 3.0.3
...
+    | Airflow version | Description |
+    |-|-|
+    | v2.0 | Initial releaase |
+
+    # Trying the API
+    You can use a third party client, such as [curl](https://curl.haxx.se/), [HTTPie](https://httpie.org/),
+    [Postman](https://www.postman.com/) or the [Insomnia rest client](https://insomnia.rest/) to test
+    the Apache Airflow API.
+
+    Note that you will need to pass an credentials data.
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
    Note that you will need to pass credentials data.
```

File: airflow/api_connexion/openapi/v1.yaml [hunk 5]
```diff
@@ -19,7 +19,148 @@ openapi: 3.0.3
...
+
+    For e.g., here is how to pause a DAG with [curl](https://curl.haxx.se/), when basic authroization is used:
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
    For e.g., here is how to pause a DAG with [curl](https://curl.haxx.se/), when basic authorization is used:
```

File: airflow/api_connexion/openapi/v1.yaml [hunk 6]
```diff
@@ -19,7 +19,148 @@ openapi: 3.0.3
...
+      - In *Postman*, you can click the **import** button at the top
+      - With *Insomnia*, you can just drag-and-drop the file on the UI
+
+    Note that with *Postman*, you can also generate code snippets by selecting a request and clicking on
+    the **Code** button.
+
+    # Authentication
+
+    To be able to meet the requirements of many organizations, Airflow supports many authentication methods,
+    and it is even possible to add your own by the most demanding users.
```
Comment (kaxil) [inline airflow/api_connexion/openapi/v1.yaml]:
```suggestion
    To be able to meet the requirements of many organizations, Airflow supports many authentication methods,
    and it is even possible to add your own method.
```

**Review by kaxil — COMMENTED** [review]
Minor suggestions or else LGTM

**Comment by mik-laj** [comment]
@kaxil All suggestion accepted.

**Review by kaxil — APPROVED** [review]

**Pull request merged**